ambry-store/src/main/java/com/github/ambry/store/BlobStoreCompactor.java

```diff
@@ -556,27 +530,15 @@ private boolean copyDataByLogSegment(LogSegment logSegmentToCopy, FileSpan dupli
   private boolean copyDataByIndexSegment(LogSegment logSegmentToCopy, IndexSegment indexSegmentToCopy,
       FileSpan duplicateSearchSpan) throws IOException, StoreException {
     logger.debug("Copying data from {}", indexSegmentToCopy.getFile());
-    List<IndexEntry> allIndexEntries = new ArrayList<>();
     // call into diskIOScheduler to make sure we can proceed (assuming it won't be 0).
     diskIOScheduler.getSlice(INDEX_SEGMENT_READ_JOB_NAME, INDEX_SEGMENT_READ_JOB_NAME, 1);
-    // get all entries. We get one entry per key
-    indexSegmentToCopy.getIndexEntriesSince(null, new FindEntriesCondition(Long.MAX_VALUE), allIndexEntries,
-        new AtomicLong(0), true);
-
-    // save a token for restart (the key gets ignored but is required to be non null for construction)
```

Comment:
Can you explain why this logic is moved?

Reply:
It's not removed, it's relocated, to the method `getValidIndexEntries`.